modular_zubbers/code/modules/changeling_zombies/items.dm

```diff
@@ -29,49 +29,40 @@
 	else
 		. += span_warning("Has a [blood_chance]% chance to infect when drawing blood on hit, with a [CHANGELING_ZOMBIE_REINFECT_DELAY/10] second cooldown.")
 
-/obj/item/melee/arm_blade/changeling_zombie/attack(mob/living/target_mob, mob/living/user, params)
+/obj/item/melee/arm_blade/changeling_zombie/attack(mob/living/target_mob, mob/living/user, list/modifiers, list/attack_modifiers)
 	. = ..()
-	if(COOLDOWN_FINISHED(src,sound_cooldown) && prob(length(attack_living_sounds)*25))
-		playsound(src,pick(attack_living_sounds),50,TRUE,SOUND_RANGE*2)
+	if(COOLDOWN_FINISHED(src, sound_cooldown) && prob(length(attack_living_sounds) * 25))
+		playsound(src, pick(attack_living_sounds), 50, TRUE, SOUND_RANGE * 2)
 		COOLDOWN_START(src, sound_cooldown, 3 SECONDS)
+	if(.) // TRUE means we were blocked
+		return
+	if(blood_chance <= 0 || target_mob.stat == DEAD || !user) //Can't infect. Will still draw blood anyways.
+		return
+	var/final_force = CALCULATE_FORCE(src, attack_modifiers)
+	if(final_force <= 0 || !ishuman(target_mob) || !COOLDOWN_FINISHED(src, infection_cooldown))
+		return
+	COOLDOWN_START(src, infection_cooldown, CHANGELING_ZOMBIE_REINFECT_DELAY)
+	if(!prob(blood_chance))
+		target_mob.balloon_alert(user, "chance to infect failed")
+		return
+	if(!can_become_changeling_zombie(target_mob))
+		target_mob.balloon_alert(user, "cannot infect")
+		return
+	var/mob/living/carbon/human/host = target_mob
+	var/datum/component/changeling_zombie_infection/infection = host.AddComponent(/datum/component/changeling_zombie_infection)
+	if(infection)
+		target_mob.balloon_alert(user, "infected")
+		var/datum/component/changeling_zombie_infection/component = user.GetComponent(/datum/component/changeling_zombie_infection)
+		infection.was_changeling_husked = component.was_changeling_husked
+		if(component && component.infect_objective)
```

Comment:
Should move the ``if(component)`` check earlier so it is before the ``infection.was_changeling_husked = component.was_changeling_husked`` line